interactions/api/models/channel.py

```diff
@@ -1231,13 +1231,9 @@ async def get_permissions_for(self, member: "Member") -> Permissions:
         if not self.guild_id:
             return Permissions.DEFAULT
 
-        from .guild import Guild
+        permissions = await member.get_guild_permissions(self.guild_id)
```

Comment:
Unsure but I think channel has guild id only in certain events, have you checked if this exists when you get the channel manually?

Reply:
If `guild_id` is None, it returns default permissions (lines above). Unsure if that's what you meant?
But yes I have tried it by getting the channel and member, and running `channel.get_permissions_for(member)`

Reply:
Oh I didn't see that, Nevermind then